File: Dec2020/14Dec/BobAndWalls.java
/*
Problem Name: Bob and Walls
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/bob-and-walls/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt(), last = 0, cnt = 0;
            for(int i=0;i<n;i++){
                int h = in.nextInt();
                if(h > last){
                    cnt++;
                    last = h;
                }
            }
            out.println(cnt);
        }
        out.flush();
    }
}

File: Dec2020/14Dec/BobAndWalls.cpp
/*
Problem Name: Bob and Walls
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/bob-and-walls/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    test(t){
        int n, h, last, counted = 1;
        cin>>n;
        cin>>h;
        last=h;
        for(int i=1;i<n;++i){
            cin>>h;
            if(h > last){
                counted++;
                last=h;
            }
        }
        cout<<counted<<"\n";
    }
    return 0;
}
